engine/src/main/java/org/terasology/rendering/dag/FinalPostProcessingNode.java

```diff
@@ -0,0 +1,226 @@
+import static org.lwjgl.opengl.GL11.GL_COLOR_BUFFER_BIT;
+import static org.lwjgl.opengl.GL11.GL_DEPTH_BUFFER_BIT;
+import static org.lwjgl.opengl.GL11.glClear;
+import static org.terasology.rendering.opengl.OpenGLUtils.bindDisplay;
+import static org.terasology.rendering.opengl.OpenGLUtils.renderFullscreenQuad;
+import static org.terasology.rendering.opengl.OpenGLUtils.setViewportToSizeOf;
+
+/**
+ * TODO: Add diagram of this node
+ * TODO: Break into two different nodes
+ */
+public class FinalPostProcessingNode implements Node {
+
+    @In
+    private WorldRenderer worldRenderer;
+
+    @In
+    private Config config;
+
+    @In
+    private PostProcessor postProcessor;
+
+    @In
+    private FrameBuffersManager frameBuffersManager;
+
+    private RenderingDebugConfig renderingDebugConfig;
+    private RenderingConfig renderingConfig;
+
+    private FBO.Dimensions fullScale;
+
+    private Material finalPost;
+    private Material debug;
+    private Material ocDistortion;
+
+    private FBO sceneFinal;
+    private FBO ocUndistorted;
+    private FBO sceneOpaque;
+
+    @Override
+    public void initialise() {
+        renderingConfig = config.getRendering();
+        renderingDebugConfig = renderingConfig.getDebug();
+
+        ocDistortion = worldRenderer.getMaterial("engine:prog.ocDistortion");
+        finalPost = worldRenderer.getMaterial("engine:prog.post"); // TODO: rename shader to finalPost
+        debug = worldRenderer.getMaterial("engine:prog.debug");
+    }
+
+    /**
+     * If each is enabled through the rendering settings, this method
+     * adds depth-of-field blur, motion blur and film grain to the rendering
+     * obtained so far. If OculusVR support is enabled, it composes (over two
+     * calls) the images for each eye into a single image, and applies a distortion
+     * pattern to each, to match the optics in the OculusVR headset.
+     * <p>
+     * Finally, it either sends the image to the display or, when taking a screenshot,
+     * instructs the FrameBuffersManager to save it to a file.
+     * <p>
+     * worldRenderer.getCurrentRenderStage() Can be MONO, LEFT_EYE or RIGHT_EYE, and communicates to the method weather
+     * it is dealing with a standard display or an OculusVR setup, and in the
+     * latter case, which eye is currently being rendered. Notice that if the
+     * OculusVR support is enabled, the image is sent to screen or saved to
+     * file only when the value passed in is RIGHT_EYE, as the processing for
+     * the LEFT_EYE comes first and leads to an incomplete image.
+     */
+    @Override
+    public void process() {
+        PerformanceMonitor.startActivity("rendering/finalpostprocessing");
+
+        ocUndistorted = frameBuffersManager.getFBO("ocUndistorted");
+        sceneFinal = frameBuffersManager.getFBO("sceneFinal");
+        sceneOpaque = frameBuffersManager.getFBO("sceneOpaque");
+
+        fullScale = sceneOpaque.dimensions();
```

Comment:
I am somewhat concerned that all these variables are reset here rather than close to where they are used. But I'm ok with leaving them as they are, as we will handle buffers and dimensions very differently, eventually.